### PR TITLE
[#171642423] Fetch testenv2 metadata

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/utils/middleware.ts [hunk 1]
```diff
@@ -80,6 +79,11 @@ export const getSpidStrategyOptionsUpdater = (
     )
     .concat([
       fetchIdpsMetadata(serviceProviderConfig.spidCieUrl, CIE_IDP_IDENTIFIERS)
+    ])
+    .concat([
```
Comment (gunzip) [inline src/utils/middleware.ts]: spidTestentUrl must be optional so I'm expeting here a check as for validator:

`NonEmptyString.is(serviceProviderConfig.spidTesetenvUrl)`

Is it possible to make spidCieUrl optional as well ? I've changed my mind
(this makes this package reusable in context where there's no need to have CIE login)

Comment (BurnedMarshal) [reply src/utils/middleware.ts]: Optional only inside `io-spid-commons` or even on `io-backend`?

Comment (BurnedMarshal) [reply src/utils/middleware.ts]: fixed in 831de37

Comment (gunzip) [reply src/utils/middleware.ts]: io-spid-commons (isn't the same ? I mean, the IServiceproviderConfig)

**Comment by BurnedMarshal** [comment]
# [Codecov](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=h1) Report
> Merging [#23](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=desc) into [master](https://codecov.io/gh/pagopa/io-spid-commons/commit/28a78153b67545c70935528a2cd4a2b205c3541b&el=desc) will **decrease** coverage by `0.17%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #23      +/-   ##
==========================================
- Coverage   64.29%   64.11%   -0.18%     
==========================================
  Files          10        9       -1     
  Lines         605      602       -3     
  Branches       88       86       -2     
==========================================
- Hits          389      386       -3     
  Misses        215      215              
  Partials        1        1              
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=footer). Last update [28a7815...831de37](https://codecov.io/gh/pagopa/io-spid-commons/pull/23?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/utils/middleware.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/utils/middleware.ts]: I mean that into `io-backend` the ENV vars for `CIE_METADATA_URL` is required and  `SPID_TESTENV_URL` if missing use default value.

Comment (gunzip) [reply src/utils/middleware.ts]: in the backend it's ok ta maintain the current approach

**Pull request merged**